app/api/server/v1/im.js

```diff
@@ -146,7 +146,8 @@ API.v1.addRoute(['dm.files', 'im.files'], { authRequired: true }, {
 		const { offset, count } = this.getPaginationItems();
 		const { sort, fields, query } = this.parseJsonQuery();
 
-		const ourQuery = Object.assign({}, query, { rid: findResult.room._id });
+		// thumbnails don't need to be fetched);
+		const ourQuery = Object.assign({}, query, { rid: findResult.room._id, name: new RegExp('^(?!thumb-).+') });
```

Comment:
what if the *full* file name actually is `thumb-something.jpg`? we need a better query to filter thumb files.

Reply:
yeah, I thought about that ... but is there a way to query for thumbnails without adding new fields to files?

Reply:
looking at the fields added to thumb files, looks like there is currently no good fields to filter:
https://github.com/RocketChat/Rocket.Chat/blob/13e0983ce2fda48a51f42d9ba9fd5a096d548f07/app/file-upload/server/lib/FileUpload.js#L296

we might need to add a field specifying those files are related to some other file, so we don't need to list them.